src/components/account/AccountCreateFirst.tsx

```diff
@@ -1,25 +1,39 @@
 import { useCallback, useEffect } from 'react'
 import { useLocation, useNavigate, useSearchParams } from 'react-router-dom'
 
+import AccountFundContent from 'components/account/AccountFund/AccountFundContent'
 import AccountFundFullPage from 'components/account/AccountFund/AccountFundFullPage'
+import Card from 'components/common/Card'
+import { CircularProgress } from 'components/common/CircularProgress'
 import FullOverlayContent from 'components/common/FullOverlayContent'
 import WalletSelect from 'components/Wallet/WalletSelect'
-import useToggle from 'hooks/common/useToggle'
 import useStore from 'store'
+import useAccounts from 'hooks/accounts/useAccounts'
+import useCurrentAccount from 'hooks/accounts/useCurrentAccount'
+import { useWeb3WalletConnection } from 'hooks/wallet/useWeb3WalletConnections'
+import useToggle from 'hooks/common/useToggle'
+import useChainConfig from 'hooks/chain/useChainConfig'
 import { getPage, getRoute } from 'utils/route'
 
 export default function AccountCreateFirst() {
   const navigate = useNavigate()
   const { pathname } = useLocation()
+  const [searchParams] = useSearchParams()
   const address = useStore((s) => s.address)
   const createAccount = useStore((s) => s.createAccount)
   const [isCreating, setIsCreating] = useToggle(false)
-  const [searchParams] = useSearchParams()
+  const chainConfig = useChainConfig()
+
+  const { data: accounts, isLoading } = useAccounts('default', address)
+  const currentAccount = useCurrentAccount()
+  const { handleConnectWallet } = useWeb3WalletConnection()
 
   useEffect(() => {
     if (!address) useStore.setState({ focusComponent: { component: <WalletSelect /> } })
   }, [address])
 
+  const hasExistingAccount = accounts && accounts.length > 0
```

Comment:
This should be a useMemo
```suggestion
  const hasExistingAccount = useMemo(() => accounts && accounts.length > 0, [accounts])
```